clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -86,6 +86,18 @@ public ProducerRecord(String topic, Integer partition, K key, V value) {
     }
 
     /**
+     * Creates a record to be sent to Kafka with a specific timestamp
+     *
+     * @param topic The topic the record will be appended to
+     * @param timestamp The timestamp of the record
+     * @param key The key that will be included in the record
+     * @param value The record contents
+     */
+    public ProducerRecord(String topic, Long timestamp, K key, V value) {
```

Comment:
Adding this constructor is quite dangerous because it's almost the same as the one that takes `partition` (the only difference is that one is a `Long` and the other is an `Integer`.